src/Components/ServiceSelectionScene/ServiceSelectionScene.tsx

```diff
@@ -108,6 +136,96 @@ export class ServiceSelectionScene extends SceneObjectBase<ServiceSelectionScene
         }
       })
     );
+
+    if (this.isTimeRangeTooEarlyForAggMetrics()) {
+      this.onUnsupportedAggregatedMetricTimeRange();
+      if (this.state.$data.state.data?.state !== LoadingState.Done) {
+        this.runServiceQueries();
+      }
+    } else {
+      this.onSupportedAggregatedMetricTimeRange();
+      if (this.state.$data.state.data?.state !== LoadingState.Done) {
+        this.runServiceQueries();
+      }
+    }
+
+    // Update labels on time range change
+    this._subs.add(
+      sceneGraph.getTimeRange(this).subscribeToState(() => {
+        if (this.isTimeRangeTooEarlyForAggMetrics()) {
+          this.onUnsupportedAggregatedMetricTimeRange();
+        } else {
+          this.onSupportedAggregatedMetricTimeRange();
+        }
+        this.runServiceQueries();
+      })
+    );
+
+    // Update labels on datasource change
+    this._subs.add(
+      getDataSourceVariable(this).subscribeToState(() => {
+        this.runServiceQueries();
+      })
+    );
+
+    if (aggregatedMetricsEnabled) {
+      this._subs.add(
+        this.getQueryOptionsToolbar()?.subscribeToState((newState, prevState) => {
+          if (newState.options.aggregatedMetrics.userOverride !== prevState.options.aggregatedMetrics.userOverride) {
+            this.runServiceQueries();
+          }
+        })
+      );
+    }
+  }
+
+  private isTimeRangeTooEarlyForAggMetrics(): boolean {
+    const timeRange = sceneGraph.getTimeRange(this);
+    return timeRange.state.value.from.isBefore(dateTime(AGGREGATED_METRIC_START_DATE));
+  }
+
+  private onUnsupportedAggregatedMetricTimeRange() {
+    const toolbar = this.getQueryOptionsToolbar();
+    toolbar?.setState({
+      options: {
+        aggregatedMetrics: {
+          ...toolbar?.state.options.aggregatedMetrics,
+          disabled: true,
+        },
+      },
+    });
+  }
+
+  private getQueryOptionsToolbar() {
+    const indexScene = sceneGraph.getAncestor(this, IndexScene);
+    return indexScene.state.controls.find((control) => control instanceof ToolbarScene) as ToolbarScene | undefined;
+  }
+
+  private onSupportedAggregatedMetricTimeRange() {
+    const toolbar = this.getQueryOptionsToolbar();
+    toolbar?.setState({
+      options: {
+        aggregatedMetrics: {
+          ...toolbar?.state.options.aggregatedMetrics,
+          disabled: false,
+        },
+      },
+    });
+  }
+
+  private runServiceQueries() {
+    const toolbar = this.getQueryOptionsToolbar();
+    const toolbarEnabled =
+      !toolbar?.state.options.aggregatedMetrics.disabled && toolbar?.state.options.aggregatedMetrics.active;
+
+    if ((!this.isTimeRangeTooEarlyForAggMetrics() || !aggregatedMetricsEnabled) && toolbarEnabled) {
```

Comment:
I know naming this is not easy, but maybe something like `aggregatedMetricsActive` instead of `toolbarEnabled`?

```suggestion
    const aggregatedMetricsActive =
      !toolbar?.state.options.aggregatedMetrics.disabled && toolbar?.state.options.aggregatedMetrics.active;

    if ((!this.isTimeRangeTooEarlyForAggMetrics() || !aggregatedMetricsEnabled) && aggregatedMetricsActive) {
```